### ISSUE TITLE
Accessing object property using indexer notation

### ISSUE THREAD

**Issue by ebekker** [lunet-io/scriban#116]
Based on [the docs](https://github.com/lunet-io/scriban/blob/master/doc/language.md#5-objects) I was under the impression I can access a property of an object using either the simple _dot-notation_ or the _indexer-notation_, e.g. if I have an object `foo` with a property `name`, I thought both of these are legal:

```
{{ foo.name }}
{{ foo["name"] }}
```

But a sample of my code that uses the indexer notation fails with this exception:

```
Unhandled Exception: Scriban.Syntax.ScriptRuntimeException: Expecting a list. Invalid value `Foo.Model.Attribute/Attribute` for the target `att` for the indexer: att[key]
```

Where `att` is a global variable of the type `Foo.Model.Attribute` -- a native .NET type.

Do I misunderstand the usage?

The reason I was trying to resort to the index-notation is that the `Attribute` type has a property named `Null` which surfaces as `null` in the scriban template and seemingly causes a conflict with the reserved keyword of the same name.  For example, both of these are failing:

```
{{ if att.null }}Not Required{{ end }}

{{ if att["null"] }}Not Required{{ end }}
```

**Comment by ebekker** [comment]
BTW, I'm using what I believe is the latest stable version of the library, v1.2.9.

**Comment by xoofx** [comment]
Don't remember, but it might work only for ScriptObject

**Comment by ebekker** [comment]
If that's the case, is that by design?

**Comment by xoofx** [comment]
> If that's the case, is that by design?

I don't remember 😉 (currently don't have time to look into the code)

**Comment by xoofx** [comment]
Should be fixed by commit 548998c that will be part of `2.0.1+`
